app/svelte/src/client/docs/sourceDecorator.ts

```diff
@@ -160,12 +160,12 @@ export const sourceDecorator = (storyFn: any, context: StoryContext<AnyFramework
     return story;
   }
 
-  const { parameters = {}, args = {} } = context || {};
+  const { parameters = {}, args = {}, component: ctxtComponent } = context || {};
   let { Component: component = {} } = story;
 
   const { wrapper, slotProperty } = getWrapperProperties(component);
   if (wrapper) {
-    component = parameters.component;
+    component = parameters.component ?? ctxtComponent;
```

Comment:
I had a look at this with some other maintainers, and as far we can tell, grabbing the component off of `parameters` should have never been supported.

What if we change this to:

```suggestion
    component = ctxtComponent;
```

And add a deprecation warning if `parameters.component` is defined? For that, we use the [`util-deprecate` package](https://www.npmjs.com/package/util-deprecate) throughout the project. ([example usages](https://github.com/storybookjs/storybook/search?q=import+deprecate))

Reply:
I agree with you. I don't remember why I used 'parameters.component' in the first place.. I updated my PR with a deprecation warning.